RecoLocalTracker/SiPixelClusterizer/plugins/alpaka/SiPixelRawToClusterKernel.dev.cc

```diff
@@ -0,0 +1,810 @@
+    //http://cmslxr.fnal.gov/source/CondFormats/SiPixelObjects/src/PixelROC.cc?v=CMSSW_9_2_0#0071
+    // Convert local pixel to pixelDetails::global pixel
+    ALPAKA_FN_ACC ::pixelDetails::Pixel frameConversion(
+        bool bpix, int side, uint32_t layer, uint32_t rocIdInDetUnit, ::pixelDetails::Pixel local) {
+      int slopeRow = 0, slopeCol = 0;
+      int rowOffset = 0, colOffset = 0;
+
+      if (bpix) {
+        if (side == -1 && layer != 1) {  // -Z side: 4 non-flipped modules oriented like 'dddd', except Layer 1
+          if (rocIdInDetUnit < 8) {
+            slopeRow = 1;
+            slopeCol = -1;
+            rowOffset = 0;
+            colOffset = (8 - rocIdInDetUnit) * ::pixelDetails::numColsInRoc - 1;
+          } else {
+            slopeRow = -1;
+            slopeCol = 1;
+            rowOffset = 2 * ::pixelDetails::numRowsInRoc - 1;
+            colOffset = (rocIdInDetUnit - 8) * ::pixelDetails::numColsInRoc;
+          }       // if roc
+        } else {  // +Z side: 4 non-flipped modules oriented like 'pppp', but all 8 in layer1
+          if (rocIdInDetUnit < 8) {
+            slopeRow = -1;
+            slopeCol = 1;
+            rowOffset = 2 * ::pixelDetails::numRowsInRoc - 1;
+            colOffset = rocIdInDetUnit * ::pixelDetails::numColsInRoc;
+          } else {
+            slopeRow = 1;
+            slopeCol = -1;
+            rowOffset = 0;
+            colOffset = (16 - rocIdInDetUnit) * ::pixelDetails::numColsInRoc - 1;
+          }
+        }
+
+      } else {             // fpix
+        if (side == -1) {  // pannel 1
+          if (rocIdInDetUnit < 8) {
+            slopeRow = 1;
+            slopeCol = -1;
+            rowOffset = 0;
+            colOffset = (8 - rocIdInDetUnit) * ::pixelDetails::numColsInRoc - 1;
+          } else {
+            slopeRow = -1;
+            slopeCol = 1;
+            rowOffset = 2 * ::pixelDetails::numRowsInRoc - 1;
+            colOffset = (rocIdInDetUnit - 8) * ::pixelDetails::numColsInRoc;
+          }
+        } else {  // pannel 2
+          if (rocIdInDetUnit < 8) {
+            slopeRow = 1;
+            slopeCol = -1;
+            rowOffset = 0;
+            colOffset = (8 - rocIdInDetUnit) * ::pixelDetails::numColsInRoc - 1;
+          } else {
+            slopeRow = -1;
+            slopeCol = 1;
+            rowOffset = 2 * ::pixelDetails::numRowsInRoc - 1;
+            colOffset = (rocIdInDetUnit - 8) * ::pixelDetails::numColsInRoc;
+          }
+
+        }  // side
+      }
+
+      uint32_t gRow = rowOffset + slopeRow * local.row;
+      uint32_t gCol = colOffset + slopeCol * local.col;
+      ::pixelDetails::Pixel global = {gRow, gCol};
+      return global;
+    }
+
+    ALPAKA_FN_ACC uint8_t conversionError(uint8_t fedId, uint8_t status, bool debug = false) {
+      uint8_t errorType = 0;
+
+      switch (status) {
+        case (1): {
```

Comment:
Maybe drop the (unnecessary) parentheses
```suggestion
        case 1: {
```
?